python/prairielearn.py

```diff
@@ -759,7 +777,7 @@ def string_fraction_to_number(a_sub, allow_fractions=True, allow_complex=True):
             if not np.isfinite(a_sub_parsed):
                 raise ValueError('The submitted answer is not a finite number.')
             value = a_sub_parsed
-            data['submitted_answers'] = to_json(value)
+            data['submitted_answers'] = value
```

Comment:
Why was the `to_json()` call removed here? Is `value` already in JSON format?

Reply:
It's actually the opposite. This element was saving numpy floats natively because this to_json call wasn't doing anything. So adding the new encoding to this function actually broke some tests that were using custom grader code.

Reply:
Given this, it looks like the docs for this function are now out of date:

> On successful parsing, "data" will contain a 'submitted_answers' key that is the JSON encoded parsed answer.

Reply:
Actually, looking at this function I think that the input string can be a complex number as well. I might try and see if there's a way to avoid changing this function. A little weird that sometimes it saves the submitted answer as a JSON dict with more detailed information and sometimes just an integer. Maybe the best way is to just update the custom grader tests?

Reply:
@coatless after thinking about this some, do you think we should have the numpy encodings have an opt-out option (similar to the way the dataframe element is set up), to avoid issues with breaking peoples custom grader code relying on the way that numpy integers are stored? That way, the option can be applied to the `string_fraction_to_number` function, which will avoid breaking people's custom grader code.